### PR TITLE
Add Woble unit test

### PR DESCRIPTION
**Pull request by yunhanw-google**

### CODE REVIEW

**Comment by jaylogue** [comment]
Yunhan, can you revert the changes to the *.in files that aren't strictly required for your change.

**Comment by yunhanw-google** [comment]
@jaylogue Done, thanks

**Comment by yunhanw-google** [comment]
@robszewczyk Done, thanks

File: src/test-apps/TestWoble.cpp [hunk 1]
```diff
@@ -0,0 +1,501 @@
...
+    uint8_t * data;
+
+    woble.Init(NULL, false);
+
+    WeaveLogDetail(Ble, "Start HandleCharacteristicReceivedOnePacket Woble State:");
+    woble.LogState();
+
+    first_packet = PacketBuffer::New(5);
+
+    if (first_packet->AvailableDataLength() < 5)
```
Comment (robszewczyk) [inline src/test-apps/TestWoble.cpp]: NL_TEST_ASSERT

File: src/test-apps/TestWoble.cpp [hunk 2]
```diff
@@ -0,0 +1,501 @@
...
+    {
+        PacketBuffer::Free(first_packet);
+        ExitNow();
+    }
+
+    first_packet->SetDataLength(5, NULL);
+    data = first_packet->Start();
+    if (data == NULL)
```
Comment (robszewczyk) [inline src/test-apps/TestWoble.cpp]: NL_TEST_ASSERT

File: src/test-apps/TestWoble.cpp [hunk 3]
```diff
@@ -0,0 +1,501 @@
...
+    }
+
+    data[0] = 0x05;
+    data[1] = 1;
+    data[2] = 1;
+    data[3] = 0;
+    data[4] = 0xff; // payload
+
+    err = woble.HandleCharacteristicReceived(first_packet, rcvd_ack, did_rcv_ack);
+    first_packet = NULL;
```
Comment (robszewczyk) [inline src/test-apps/TestWoble.cpp]: NL_TEST_ASSERT

File: src/test-apps/TestWoble.cpp [hunk 4]
```diff
@@ -0,0 +1,501 @@
...
+    SuccessOrExit(err);
+
+    VerifyOrExit(woble.RxState() == WoBle::kState_InProgress, err = BLE_ERROR_INCORRECT_STATE);
+
+    second_packet = PacketBuffer::New(3);
+    second_packet->SetDataLength(3, NULL);
+    data = second_packet->Start();
+    if (data == NULL)
```
Comment (robszewczyk) [inline src/test-apps/TestWoble.cpp]: NL_TEST_ASSERT

File: src/test-apps/TestWoble.cpp [hunk 5]
```diff
@@ -0,0 +1,501 @@
...
+        PacketBuffer::Free(second_packet);
+        ExitNow();
+    }
+
+    data[0] = WoBle::kHeaderFlag_EndMessage;
+    data[1] = 2;
+    data[4] = 0xff; // payload
+
+    err = woble.HandleCharacteristicReceived(second_packet, rcvd_ack, did_rcv_ack);
+    second_packet = NULL;
```
Comment (robszewczyk) [inline src/test-apps/TestWoble.cpp]: NL_TEST_ASSERT

File: src/test-apps/TestWoble.cpp [hunk 6]
```diff
@@ -0,0 +1,501 @@
...
+exit:
+    if (first_packet != NULL)
+    {
+        PacketBuffer::Free(first_packet);
+    }
+
+    data = NULL;
+    woble.Init(NULL, false);
+
+    WeaveLogDetail(Ble, "UnitTest6 with Woble State:");
```
Comment (robszewczyk) [inline src/test-apps/TestWoble.cpp]: Nit: still logs UnitTest6

Comment (yunhanw-google) [reply src/test-apps/TestWoble.cpp]: Fixed. thanks

File: src/test-apps/TestWoble.cpp (hunk 5 above)
Comment (yunhanw-google) [reply src/test-apps/TestWoble.cpp]: Done, thanks

File: src/test-apps/TestWoble.cpp (hunk 3 above)
Comment (yunhanw-google) [reply src/test-apps/TestWoble.cpp]: Done, thanks

File: src/test-apps/TestWoble.cpp [hunk 7]
```diff
@@ -0,0 +1,393 @@
...
+    NL_TEST_ASSERT(inSuite, data != NULL);
+
+    data[0] = 0x05;
+    data[1] = 1;
+    data[2] = 1;
+    data[3] = 0;
+    data[4] = 0xff; // payload
+
+    err = woble.HandleCharacteristicReceived(first_packet, rcvd_ack, did_rcv_ack);
+    first_packet = NULL;
```
Comment (robszewczyk) [inline src/test-apps/TestWoble.cpp]: Please replace `PacketBuffer::New()` with `PacketBuffer::NewWithAvailableSize()`

File: src/test-apps/TestWoble.cpp [hunk 8]
```diff
@@ -0,0 +1,393 @@
...
+    NL_TEST_ASSERT(inSuite, woble.RxState() == WoBle::kState_InProgress);
+
+    second_packet = PacketBuffer::New(3);
+    second_packet->SetDataLength(3, NULL);
+    data = second_packet->Start();
+    NL_TEST_ASSERT(inSuite, data != NULL);
+
+    data[0] = WoBle::kHeaderFlag_EndMessage;
+    data[1] = 2;
+    data[4] = 0xff; // payload
```
Comment (robszewczyk) [inline src/test-apps/TestWoble.cpp]: Typo, `4` should be `2`

**Pull request merged**